src/pages/main/collaboration/collaboration.ts

```diff
@@ -23,17 +23,34 @@ export const useCollaboration = (projectId: string): Collaboration => {
   const clientIdRef = useRef<string>(null);
   clientIdRef.current = state.clientId;
 
-  const deltaQueue = useDeltaQueue(clientIdRef, state, dispatch, sendMessage);
+  const lamportClock = useRef<number>(0);
+  
+  useEffect(() => {
+    lamportClock.current = Math.max(lamportClock.current, state.initClock);
+  }, [state.initClock]);
 
   const nextCharacterIdRef = useRef<number>(0);
 
+  const handleIncomingClock = (incomingClock: number) => 
+    lamportClock.current = Math.max(lamportClock.current, incomingClock);
```

Comment:
Maybe I don't understand what there is on Wikipedia, but shouldn't there be
```suggestion
  const handleIncomingClock = (incomingClock: number) => 
    lamportClock.current = Math.max(lamportClock.current, incomingClock) + 1;
```
?

Reply:
would work too, but every time I create new character i do

```
lamportClock++;
char.clock = lamportClock
```

So it also fine the way it is

Reply:
Okay, if you say so